core/src/main/java/org/apache/calcite/rel/rules/PruneEmptyRules.java

```diff
@@ -402,12 +404,8 @@ private static boolean isEmpty(RelNode node) {
     public void onMatch(RelOptRuleCall call) {
       SingleRel single = call.rel(0);
       RelNode emptyValues = call.builder().push(single).empty().build();
-      if (single instanceof Sort) {
-        emptyValues = emptyValues.copy(
-            emptyValues.getTraitSet().replace(((Sort) single).getCollation()),
-            Collections.emptyList());
-      }
-      call.transformTo(emptyValues);
+      RelNode result = emptyValues.copy(single.getTraitSet(), Collections.emptyList());
```

Comment:
If you change PruneEmptyRule's autoPruneOld to false, you might be able to see another error.

Reply:
That's rigth, in that case I get:
```
java.lang.AssertionError
	at org.apache.calcite.rel.logical.LogicalValues.copy(LogicalValues.java:95)
```
Which refers to the following line:
```
@Override public RelNode copy(RelTraitSet traitSet, List<RelNode> inputs) {
    assert traitSet.containsIfApplicable(Convention.NONE);  // <--
    ...
```

Reply:
Actually, we can solve the problem with a different approach, and it would take just one line of code, which is returning false in PruneEmptyRule's autoPruneOld:
```
  protected abstract static class PruneEmptyRule extends RelOptRule
      implements SubstitutionRule {
    ...
    @Override public boolean autoPruneOld() {
      return false; // this solves the problem!
    }
  }
```

Reply:
It will just hide the issue of the rule. We still get a sort in the plan.